cmd/web-handlers.go

```diff
@@ -818,3 +789,93 @@ func presignedGet(host, bucket, object string) string {
 	// Construct the final presigned URL.
 	return host + path + "?" + query + "&" + "X-Amz-Signature=" + signature
 }
+
+// toJSONError converts regular errors into more user friendly
+// and consumable error message for the browser UI.
+func toJSONError(err error, params ...string) (jerr *json2.Error) {
+	apiErr := toWebAPIError(err)
+	jerr = &json2.Error{
+		Message: apiErr.Description,
+	}
+	switch apiErr.Code {
+	// Bucket name invalid with custom error message.
+	case "InvalidBucketName":
+		if len(params) > 0 {
+			jerr = &json2.Error{
+				Message: fmt.Sprintf("Bucket Name %s is invalid. Lowercase letters, period and numerals are the only allowed characters.",
```

Comment:
Not sure how to word it, but the S3 restrictions around period handling (in bucket names) might need to be referenced too.  For example, if someone creates a bucket with the first or last character being a period - an invalid name - this error message wouldn't help explain what's wrong.

Is there any way to include a link to the docs for further info?  eg: "Further information: https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html#bucketnamingrules"